components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -638,12 +579,17 @@ void AdsServiceImpl::OnInitialize(
     is_initialized_ = true;
 
     // OnBraveAdsServiceReady is implemented for Android for now
-    #if defined(OS_ANDROID)
-// TODO(jocelyn): FIXME
-//      auto* display_service_impl =
-//          static_cast<NotificationDisplayServiceImpl*>(display_service_);
-//      display_service_impl->OnBraveAdsServiceReady(is_initialized_);
-    #endif
+// #if defined(OS_ANDROID)
+//    auto* display_service_impl =
+//        static_cast<NotificationDisplayServiceImpl*>(display_service_);
+//    display_service_impl->OnBraveAdsServiceReady(is_initialized_);
+// #endif
+
+    auto* ads_notification_handler = static_cast<AdsNotificationHandler*>(
+        profile_->GetUserData(AdsNotificationHandler::UserDataKey()));
+    if (ads_notification_handler) {
```

Comment:
this should be a `CHECK` because it should never happen

Reply:
same for https://github.com/brave/brave-core/pull/4093/files#diff-25f5c78ce876d62668d65cb9ddcb443dR555